From 432a29090c061b8567d69c576f6761904c8d228b Mon Sep 17 00:00:00 2001
From: Rasmus Thomsen <rasmus.thomsen@protonmail.com>
Date: Sun, 20 Jan 2019 02:09:53 +0100
Subject: [PATCH] meson: ensure correct build order of VAPI

Without this commit spice-client-gtk-3.0.vapi may be built
before spice-client-glib-2.0.vapi if build_jobs > 1. This causes
the build to fail because the former depends on the latter
---
 vapi/meson.build | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/vapi/meson.build b/vapi/meson.build
index 2c4caa0..c9fef3d 100644
--- vapi/meson.build
+++ vapi/meson.build
@@ -1,12 +1,13 @@
 if spice_gtk_has_vala
-  gnome.generate_vapi('spice-client-glib-2.0',
+  spice_glib_vapi = gnome.generate_vapi('spice-client-glib-2.0',
                       install : true,
                       packages : ['gio-2.0', 'gstreamer-1.0'],
                       sources : spice_client_glib_gir[0])
+
   if spice_gtk_has_gtk
     gnome.generate_vapi('spice-client-gtk-3.0',
                         install : true,
-                        packages : ['gtk+-3.0', 'gstreamer-1.0', 'spice-client-glib-2.0'],
+                        packages : ['gtk+-3.0', 'gstreamer-1.0', spice_glib_vapi],
                         gir_dirs : join_paths(meson.build_root(), 'src'),
                         vapi_dirs : meson.current_build_dir(),
                         sources : spice_client_gtk_gir[0])
-- 
2.20.1

